src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSCopy.java

```diff
@@ -57,8 +62,36 @@ public void run(ActionContext context) throws IOException {
     }
     StorageClient storageClient = StorageClient.create(config.getProject(), config.getServiceAccount(),
                                                        isServiceAccountFilePath);
-    //noinspection ConstantConditions
-    storageClient.copy(config.getSourcePath(), config.getDestPath(), config.recursive, config.shouldOverwrite());
+
+    GCSPath destPath = config.getDestPath();
+    GCSPath undo = null;
+    Storage storage = storageClient.storage;
+
+    // create the destination bucket if not exist
+    Bucket bucket = null;
+    try {
+      bucket = storage.get(destPath.getBucket());
+    } catch (StorageException e) {
+      throw new RuntimeException(
+        String.format("Unable to access or create bucket %s. ", destPath.getBucket())
```

Comment:
this message is wrong, it is not trying to create the bucket.

Also, is this an additional permission that we're adding (which would be backwards incompatible)? Seems like the plugin didn't need read access before, it just needed to be able to write?

Reply:
No, we don't need any additional permission to access the bucket. If it has write permission it will be able to access the bucket. Will be changing the message to "Unable to access the bucket"

Reply:
Are you sure? They look like different permissions based on https://cloud.google.com/storage/docs/access-control/iam-permissions. Did you actually try it out?